sfputil/main.py

```diff
@@ -1112,6 +1112,61 @@ def run_firmware(port_name, mode):
 
     return status
 
+def is_fw_switch_done(port_name):
+    """
+        Make sure the run_firmware cmd is done
+        @port_name:
+        Returns 1 on success, and exit_code = -1 on failure
+    """
+    status = 0
+    physical_port = logical_port_to_physical_port_index(port_name)
+    sfp = platform_chassis.get_sfp(physical_port)
+
+    try:
+        api = sfp.get_xcvr_api()
+    except NotImplementedError:
+        click.echo("This functionality is currently not implemented for this platform")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+
+    try:
+        MAX_WAIT = 600 # 60s timeout.
+        fw_info = api.get_module_fw_info()
+        cnt = 0
+        is_busy = 1 if (fw_info['status'] == False) and (fw_info['result'] is not None) else 0
```

Comment:
Could you show examples of  fw_info 1) when a module is busy and 2) when module shows faulty status?

Reply:
"fw_info" is returned by "get_module_fw_info()" which is in cmis.py of "sonic-platform-common". It will return 3 combination patterns of status and result as below.
  1. {['status']: False, ['result']: None} : **Cdb** features **not supported** by switch, **or wrong chkcode** got from module. Do **NOT need to recheck** in these cases.
  2. {['status']: False, ['result']: 0} : I2C NACK or other interface issue. **Recheck** module fw info **until timeout** in this case.
  3. {['status']: True, ['result']: (ImageA & B information)} : **Valid information** got from module, **go ahead** to handle the return value.

A unit test function "test_is_fw_switch_done" in sfputil_test.py will also test all these cases.